torch/_dynamo/variables/user_defined.py

```diff
@@ -510,3 +510,52 @@ def odict_getitem(self, tx, key):
         ).add_options(
             key, self
         )
+
+
+class ProcessGroupVariable(UserDefinedObjectVariable):
+    """
+    We don't want a ProcessGroup object to end up in our output graph.
+
+    But it's common for dynamo to intercept a PG that is then used to get info like
+    rank() or world_size(), as well as passed to utility functions in distributed_c10d
+    which desugar it into plain types like a ranklist and tag.
+
+    For convenience and proper guarding, we construct a variable type.
+
+    TODO: make it possible to use ProcessGroupVariable as input to simple functions
+          like _expand_group without dynamo complaining about making a proxy for it.
+          It is not a tensor-like type, and we don't want a proxy- but dynamo assumes
+          torch library functions are dealing with tensor-like types and would have proxies
+          for their args.
+    TODO: should we make this inherit VT instead of UDOV? Do we want any of the default behaviors
+          or just graph-break whenever one of our special cases is not hit?
+    """
+
+    def __init__(self, value, **kwargs):
+        super().__init__(value, **kwargs)
+
+    def as_python_constant(self):
+        return self.value
+
+    def call_method(
+        self,
+        tx,
+        name,
+        args: "List[VariableTracker]",
+        kwargs: "Dict[str, VariableTracker]",
+    ) -> "VariableTracker":
+        if name == "rank":
+            return variables.ConstantVariable(self.value.rank())
+        if name == "size":
+            return variables.ConstantVariable(self.value.size())
+
+        # TODO should this just raise unimplemented?
+        return super().call_method(tx, name, args, kwargs)
```

Comment:
No, because you are a UserDefinedObjectVariable.

If you want to raise unimplemented, the idiomatic way is to make this a top-level variable-tracker and still invoke super, which will do it for you.

Reply:
yea makes sense. i agree, we should move to being a top-level VT.